zetaclient/orchestrator/chain_activate.go

```diff
@@ -0,0 +1,248 @@
+package orchestrator
+
+import (
+	"time"
+
+	ethcommon "github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/ethclient"
+
+	btcobserver "github.com/zeta-chain/zetacore/zetaclient/chains/bitcoin/observer"
+	btcrpc "github.com/zeta-chain/zetacore/zetaclient/chains/bitcoin/rpc"
+	btcsigner "github.com/zeta-chain/zetacore/zetaclient/chains/bitcoin/signer"
+	evmobserver "github.com/zeta-chain/zetacore/zetaclient/chains/evm/observer"
+	evmsigner "github.com/zeta-chain/zetacore/zetaclient/chains/evm/signer"
+	"github.com/zeta-chain/zetacore/zetaclient/chains/interfaces"
+	"github.com/zeta-chain/zetacore/zetaclient/common"
+	"github.com/zeta-chain/zetacore/zetaclient/config"
+)
+
+// WatchEnabledChains watches for run-time chain activation and deactivation
+func (oc *Orchestrator) WatchEnabledChains() {
+	oc.logger.Std.Info().Msg("WatchChainActivation started")
+
+	ticker := time.NewTicker(common.ZetaBlockTime * 2)
+	for {
+		select {
+		case <-ticker.C:
+			oc.ActivateAndDeactivateChains()
+		case <-oc.stop:
+			oc.logger.Std.Info().Msg("WatchChainActivation stopped")
+			return
+		}
+	}
+}
+
+// ActivateAndDeactivateChains activates and deactivates chains according to chain params and config file
+//
+// The chains to be activated:
+//   - chain params flag 'IsSupported' is true AND
+//   - chain is configured in config file
+//
+// The chains to be deactivated:
+//   - chain params flag 'IsSupported' is false OR
+//   - chain is not configured in config file
+//
+// Note:
+//   - zetaclient will reload config file periodically and update in-memory config accordingly.
+//   - As an tss signer, please make sure the config file is always well configured and not missing any chain
+func (oc *Orchestrator) ActivateAndDeactivateChains() {
```

Comment:
I continue to think `UpdateActivatedChains` would be a better name as UpdateX a recurrrent terminology in the ZetaClient

Reply:
Makes sense. Let's do it then.

Reply:
https://github.com/zeta-chain/node/pull/2411/commits/e46e0e43cbea110a018ff480fa41639da7956be4